### DISCUSSION TITLE
Generator naming conventions

### DISCUSSION THREAD

**Discussion by timothystone** [jhipster/generator-jhipster#27256] [Ideas]
The generator uses a lot of naming conventions that create friction in the use of Java Style conventions like the Google Style Guide. For example, `*DTO.java` fails **builtin** defaults of Checkstyle for Google. The built-in Sun checks pass. Picking a style guide is often opinionated.

Is there an appetite to conform to a _published and opinionated_ standard that would largely change the generator templates?

**Comment by mraible** [comment]
Can't you just adjust prettier to use your preferences, then run it to reformat all your code?

**Reply by timothystone** [reply]
Thanks @mraible. I'd have to look at that. Is it dangerous for a formatter to being changing statically typed code; it's not a refactoring engine. 🤔 Maybe I'm bringing my own priors to this idea as we've largely adopted the Google Style Guide in our practice, changing one thing: line length 100 v 120. 

Checkstyle, and to a large part Prettier, is good at: "Hey, you're line length is too long. You're variable names don't follow the agreed upon standard. Fail." Prettier will fix a lot of that in `write` routines, but would I trust Prettier to refactor a method name and all the calls of that method? I don't know. Yet. I've never asked Checkstyle to do that. One can load the Checkstyle format into the IDE and format, but personally I've never asked the IDE to also refactor beyond bracing, imports, &tc across a whole project fixing 10s of thousand of formatting errors. Generally, I have to use a context action on the offending element when something needs to be refactored.

It's just a thought I had. We fixed some of our superficial issue with configuration like `dtoSuffix Dto` in the JDL. May not be worth the squeeze, or for a major version sometime in the future. I'm just a fan of a published standard over an _ad hoc_ standard.
